### PR TITLE
Feat/add alert

### PR DESCRIPTION
**Pull request by carloshrabelo**
## Infos

[Link da task](https://juntossomosmais.monday.com/boards/448855787/pulses/593514044?userId=13833677)

#### O que está sendo entregue?

Criação de Componente de Alert 

#### Evidências

Screen Shot 2020-08-12 at 14 37 37
Screen Shot 2020-08-12 at 14 37 46
Screen Shot 2020-08-12 at 14 37 54
Screen Shot 2020-08-12 at 14 38 06

### CODE REVIEW

File: packages/styles/components/Alert.module.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+.icon {
+  display: grid;
+  font-size: $iconSize;
+  height: $iconSize;
+  place-content: center;
+  position: absolute;
+  top: var(--space-sm);
+  width: $iconSize;
```
Comment (felipefialho) [inline packages/styles/components/Alert.module.scss:41]: Why we are using `display: grid` here?

File: packages/styles/components/Alert.module.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  &.with-icon {
+    padding-left: calc(#{$iconSize} + var(--space-md));
+  }
+
+  &.with-close {
+    padding-right: calc(#{$iconSize} + var(--space-md));
+  }
+
+  @each $type, $color in $colors {
```
Comment (felipefialho) [inline packages/styles/components/Alert.module.scss:25]: Nice approach 😊

File: packages/styles/components/Icon.module.scss [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @import '~@venice/styles/core';
 
 .icon {
-  fill: var(--gray-2);
+  fill: currentColor;
```
Comment (felipefialho) [inline packages/styles/components/Icon.module.scss:4]: 👍

File: packages/styles/components/Alert.module.scss (hunk 2 above)
Comment (felipefialho) [inline packages/styles/components/Alert.module.scss:25]: nice approach 😊

File: packages/styles/components/Alert.module.scss [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    &.#{$type} {
+      .icon-alert {
+        color: $color;
+      }
+    }
+  }
+}
+
+.icon {
+  display: grid;
```
Comment (felipefialho) [inline packages/styles/components/Alert.module.scss:35]: why we are using `display: grid` here?

File: packages/styles/components/Icon.module.scss (hunk 3 above)
Comment (felipefialho) [inline packages/styles/components/Icon.module.scss:4]: 👍

File: packages/venice-types/src/IAlert.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+enum type {
+  'success',
+  'danger',
+  'toast',
+  'vip',
```
Comment (felipefialho) [inline packages/venice-types/src/IAlert.ts:5]: `vip` describe a business rule, maybe we can add a most generic name here.

File: packages/styles/components/Alert.module.scss (hunk 1 above)
Comment (carloshrabelo) [reply packages/styles/components/Alert.module.scss:41]: Para alinhar os icones no box de 16px, tanto vertical quanto horizontal

File: packages/venice-types/src/IAlert.ts (hunk 5 above)
Comment (carloshrabelo) [reply packages/venice-types/src/IAlert.ts:5]: You hava sugestions to rename this?

File: packages/styles/components/Alert.module.scss (hunk 4 above)
Comment (carloshrabelo) [reply packages/styles/components/Alert.module.scss:35]: Without Grid
Screen Shot 2020-08-12 at 12 09 15

With Grid
Screen Shot 2020-08-12 at 12 09 22

Aproach with `text-align:center; line-height:16px`
Screen Shot 2020-08-12 at 12 10 10

**Comment by carloshrabelo** [comment]
> Looking at the prints it seems that the icons are misaligned in relation to the text.
> 
> Should it be aligned to the center horizontally?

Eles tem um espaçamento de 16px do topo, mas o CSS do Storybook está colocando um `line-height` no conteúdo, dando a impressão que o texto tem um espaçamento maior, mas isso pode variar de acordo com o `line-height` de cada projeto

File: packages/styles/components/Alert.module.scss (hunk 4 above)
Comment (felipefialho) [reply packages/styles/components/Alert.module.scss:35]: uhmmmm ok, but this behavior is equal in other browsers like Firefox or Safari?

File: packages/venice-types/src/IAlert.ts (hunk 5 above)
Comment (felipefialho) [reply packages/venice-types/src/IAlert.ts:5]: maybe... `star`?

Comment (carloshrabelo) [reply packages/venice-types/src/IAlert.ts:5]: featured?

File: packages/styles/components/Alert.module.scss (hunk 4 above)
Comment (carloshrabelo) [reply packages/styles/components/Alert.module.scss]: I change aproach, still working in Chome, Firefox, Safari and edge

File: packages/venice-types/src/IAlert.ts (hunk 5 above)
Comment (felipefialho) [reply packages/venice-types/src/IAlert.ts:5]: Nice! `feature` sounds good 😁

**Pull request merged**